src/backend/distributed/executor/multi_utility.c

```diff
@@ -1255,7 +1255,13 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 {
 	bool executionOK = false;
 
-	PreventTransactionChain(isTopLevel, "distributed DDL commands");
+	if (IsInTransactionChain(isTopLevel) && XactOperations != 0)
```

Comment:
Hm, on final thought, it's more clear to just leave the call here, no? We'd have to add logic to deal with the value of `isTopLevel`, which should be considered if `IsTransactionBlock` returns false, but doesn't actually matter otherwise. I think leaving this here makes sense?